NDTensors/src/lib/TypeParameterAccessors/src/unspecify_parameters.jl

```diff
@@ -0,0 +1,45 @@
+function _unspecify_parameters(type::Type)
+  return Base.typename(type).wrapper
+end
+@generated function unspecify_parameters(type_type::Type)
+  type = parameter(type_type)
+  return _unspecify_parameters(type)
+end
+
+# Like `set_parameters` but less strict, i.e. it allows
+# setting with `TypeVar` while `set_parameters` would error.
+function new_parameters(type::Type, params)
+  return to_unionall(unspecify_parameters(type){params...})
+end
+
+function _unspecify_parameter(type::Type, pos::Int)
+  !is_parameter_specified(type, pos) && return type
+  unspecified_param = parameter(unspecify_parameters(type), pos)
+  params = Base.setindex(parameters(type), unspecified_param, pos)
+  return new_parameters(type, params)
+end
+@generated function unspecify_parameter(type_type::Type, pos_type::Position)
+  type = parameter(type_type)
+  pos = parameter(pos_type)
```

Comment:
In places like this, would it be useful to use `Int` since its defined?

Reply:
We could use that, though I was trying to have these generated functions follow a generic structure, where it just preprocesses the types of the inputs into the correct form needed by the inner function in the most general way possible. Here, in principle the `Position` could be wrapping something besides an `Int`, so I think it is best to just leave it more generic/general.